core/table.ts

```diff
@@ -141,10 +141,17 @@ export interface IBigQueryOptions {
    * For more information, see our [incremental dataset docs](https://docs.dataform.co/guides/incremental-datasets).
    */
   updatePartitionFilter?: string;
+
+  /**
+   * Key-value pairs for [BigQuery labels](https://cloud.google.com/bigquery/docs/labels-intro).
+   *
+   * Both `labels: { "key": "value" }` and `labels: { key: "value" }` as part of the `bigquery` block are valid.
```

Comment:
I'm not sure this is super helpful. I would suggest: `If the label name contains special characters, e.g. hyphens, then quote its name, e.g. labels: { "label-name": "value" }.`